private-templates-service/client/src/components/AdaptiveCardPanel/AdaptiveCardPanel.tsx

```diff
@@ -24,6 +25,25 @@ interface Props {
   template: Template;
 }
 
+function retrieveStateValue(template: Template): string {
```

Comment:
I feel that we should put this function and other functions like this in a separate class inside the utils folder since we use something like this in several places.

also can we name the function something like `getLatestTemplateInstanceState` ?

Reply:
Done.